### PR TITLE
Create update, delete and change password functions

### LINKED ISSUE #60: Create update and delete pages

### PR DESCRIPTION
**Pull request by juandrie**
closes #60

### CODE REVIEW

**Review by Hubert-Vecchioli — APPROVED** [review]

File: backend/media/avatars/pong-game-ascii-code-8bit-footage-236984349_iconl.jpeg
Comment (Hubert-Vecchioli) [inline backend/media/avatars/pong-game-ascii-code-8bit-footage-236984349_iconl.jpeg:1]: 👁️ 👁️

File: backend/auth/authentication/views.py [hunk 1]
```diff
@@ -93,13 +98,15 @@ class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
     authentication_classes = [JWTAuthentication, SessionAuthentication]
-    permission_classes = [IsAuthenticated]
+    permission_classes = [IsAuthenticated, IsOwner]
 
     def get_permissions(self):
         if self.action in ["update", "destroy", "partial_update", "retrieve"]:
-            permission_classes = [IsAdminUser]
+            logger.info("Entered case IsOwner")
```
Comment (mortebrume) [inline backend/auth/authentication/views.py:105]: please remove

File: backend/auth/authentication/views.py [hunk 2]
```diff
@@ -93,13 +98,15 @@ class UserViewSet(viewsets.ModelViewSet):
...
+            permission_classes = [IsOwner]
         else:
-            permission_classes = [IsAuthenticated]
+            logger.info("Entered case IsAuthenticated")
```
Comment (mortebrume) [inline backend/auth/authentication/views.py:108]: please remove

File: frontend/static/js/check-profile.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+async function checkEmailExists(client, email, currentEmail) {
+  if (!email || email === currentEmail) {
+    console.error('Email is empty or already used.')
+    return false
+  }
+  try {
+    await ky.get(`https://auth.api.transcendence.fr/users/check-email/?email=${email}`, {
+      headers: {
+        Authorization: `Bearer ${client.token}`,
```
Comment (mortebrume) [inline frontend/static/js/check-profile.js:51]: please add ``credentials: 'include'``

File: frontend/static/js/check-profile.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+async function checkUsernameExists(client, username, currentUsername) {
+  if (!username || username === currentUsername) {
+    console.error('Username is empty or already used.')
+    return false
+  }
+  try {
+    await ky.get(`https://auth.api.transcendence.fr/users/check-username/?username=${username}`, {
+      headers: {
+        Authorization: `Bearer ${client.token}`,
```
Comment (mortebrume) [inline frontend/static/js/check-profile.js:74]: please add ``credentials: 'include'``

File: frontend/static/js/friends.js [hunk 5]
```diff
@@ -122,7 +105,7 @@ export async function declineFriendRequest(client, fromUserId) {
 export async function loadPendingFriendRequests(client) {
   const friendRequest = document.getElementById('friend-requests-section')
   try {
-    const pendingRequests = await ky.get('https://auth.api.transcendence.local/pending-friend-requests/', {
+    const pendingRequests = await ky.get('https://auth.api.transcendence.fr/pending-friend-requests/', {
       headers: { Authorization: `Bearer ${client.token}` },
```
Comment (mortebrume) [inline frontend/static/js/friends.js:109]: please add ``credentials: 'include'``

File: frontend/static/js/check-profile.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+      },
+    }).json()
+  }
+  catch (error) {
+    if (error.response && error.response.status === 409) {
+      return true
+    }
+    else {
+      console.error('Erreur lors de la vérification de l\'email:', error)
```
Comment (mortebrume) [inline frontend/static/js/check-profile.js:60]: english

File: frontend/static/js/check-profile.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        Authorization: `Bearer ${client.token}`,
+      },
+    }).json()
+  }
+  catch (error) {
+    if (error.response && error.response.status === 409) {
+      return true
+    }
+    else {
+      console.error('Erreur lors de la vérification du username:', error)
```
Comment (mortebrume) [inline frontend/static/js/check-profile.js:83]: english

**Review by mortebrume — CHANGES_REQUESTED** [review]

**Comment by mortebrume** [comment]
The rest is good, i'm not sure about the endpoint to check if an email is valid because it's a security risk, but we will see later because i don't have any better ideas.
Also please remove the jpeg.

**Review by elrichardeau — APPROVED** [review]

**Review by mortebrume — APPROVED** [review]

**Pull request merged**